code/modules/mob/living/simple_animal/hostile/pulsedemon.dm

```diff
@@ -0,0 +1,582 @@
+            var/obj/machinery/power/apc/current_apc = current_power
+            drainAPC(current_apc)
+        else if(current_power.avail() > amount_per_regen)
+            current_power.add_load(amount_per_regen)
+        else
+            health -= health_drain_rate    
+    else
+        death()
+    regular_hud_updates()
+    ..()
+
+/mob/living/simple_animal/hostile/pulse_demon/death(var/gibbed = 0)
+    ..()
+    qdel(src)
+
+/mob/living/simple_animal/hostile/pulse_demon/proc/is_under_tile()
+    var/turf/simulated/floor/F = get_turf(src)
+    return istype(F,/turf/simulated/floor) && F.floor_tile
+
+/mob/living/simple_animal/hostile/pulse_demon/Move(NewLoc, Dir = 0, step_x = 0, step_y = 0, glide_size_override = 0)
+    if(!locate(/obj/structure/cable) in NewLoc || !locate(/obj/machinery/power) in NewLoc)
+        return
+    ..()
+    if(!is_under_tile() && prob(25))
+        spark(src,rand(2,4))
+    var/obj/machinery/power/new_power = locate(/obj/machinery/power) in NewLoc
+    if(new_power && !current_power)
+        current_power = new_power
+        current_cable = null
+        loc = new_power
+        playsound(src,'sound/weapons/electriczap.ogg',50, 1)
+        spark(src,rand(2,4))
+        if(istype(current_power,/obj/machinery/power/apc))
+            var/obj/machinery/power/apc/current_apc = current_power
+            if(current_apc.occupant)
+                to_chat(src,"<span class='warning'>Something else that isn't a pulse demon is already in here!</span>")
+                return
+            max_can_absorb = current_apc.cell.maxcharge
+            if(current_apc.pulsecompromised)
+                controlling_area = get_area(current_power)
+            else
+                hijackAPC(current_apc)
+        else if(istype(current_power,/obj/machinery/power/battery))
+            var/obj/machinery/power/battery/current_battery = current_power
+            to_chat(src,"<span class='notice'>You are now draining power from \the [current_power] and refilling charge.</span>")
+            max_can_absorb = current_battery.output
+    else
+        var/obj/structure/cable/new_cable = locate(/obj/structure/cable) in NewLoc
+        if(new_cable)
+            current_cable = new_cable
+            previous_net = current_net
+            current_net = current_cable.get_powernet()
+            current_power = null
+            current_robot = null
+            current_weapon = null
+            if(!isturf(loc))
+                loc = get_turf(NewLoc)
+            controlling_area = null
+            update_cableview()
+        else
+            current_cable = null
+            current_power = null
+            current_robot = null
+            current_weapon = null
+
+/mob/living/simple_animal/hostile/pulse_demon/to_bump(var/atom/obstacle) // Copied from how adminbus does it
+    if(!is_under_tile() && isliving(obstacle))
+        var/mob/living/L = obstacle
+        shockMob(L)
```

Comment:
mmm no cooldown shock attack

Reply:
I think you quoted the wrong thing but the unarmed_attack_mob one used to husk in 2 hits, hope I finally nerfed that.
The to_bump one really doesn't shock that often in testing.